lib/reek/ast/node.rb

```diff
@@ -102,6 +104,31 @@ def source
         loc.expression.source_buffer.name
       end
 
+      #
+      # @return [Reek::AST::Node] the condition that is associated with a conditional node.
+      #   For instance, this code
+      #
+      #     if charlie(bravo) then delta end
+      #
+      #   would be parsed into this AST:
+      #
+      #     s(:if,
+      #       s(:send, nil, :charlie,
+      #         s(:lvar, :bravo)),
+      #       s(:send, nil, :delta), nil)
+      #
+      #   so in this case we would return this
+      #
+      #     s(:send, nil, :charlie,
+      #       s(:lvar, :bravo))
+      #
+      #   as condition.
+      #
+      def condition
+        return unless CONDITIONAL_NODE_TYPES.include?(type)
+        super
```

Comment:
I think you can remove this whole body, since this method will be overridden by the code in the `IfNode`, `CaseNode`, and `LogicOperatorBase` sexp extensions.

Reply:
Unfortunately we can't remove this method because otherwise dozens of our specs will crash with errors like:

```
     # ------------------
     # --- Caused by: ---
     # NoMethodError:
     #   undefined method `condition' for #<#<Class:0x00007f82459254b8>:0x00007f82459253a0>
     #   ./lib/reek/smell_detectors/control_parameter_helpers/call_in_condition_finder.rb:47:in `uses_param_in_call_in_condition?'
```

Reply:
Not the method, just the body :slightly_smiling_face:.